file/provider/delete_volume_access_point_test.go

```diff
@@ -147,8 +147,55 @@ func TestDeleteVolumeAccessPoint(t *testing.T) {
 				assert.Nil(t, err)
 			},
 		},
+		{
+			testCaseName: "Volume Access Point deletion failed",
+			providerVolumeAccessPointRequest: provider.VolumeAccessPointRequest{
+				VolumeID:      "volume-id1",
+				AccessPointID: "16f293bf-test-4bff-816f-e199c0c65db5",
+			},
+
+			baseVolumeAccessPointResponse: &models.ShareTarget{
+				ID:        "16f293bf-test-4bff-816f-e199c0c65db5",
+				MountPath: "abac:/asdsads/asdsad",
+				Name:      "test volume name",
+				Status:    "stable",
+				VPC:       &provider.VPC{ID: "VPC-id1"},
+				ShareID:   "",
+				Zone:      &models.Zone{Name: "test-zone"},
+			},
+
+			baseHttpResponse: &http.Response{
+				StatusCode: http.StatusInternalServerError,
+			},
+
+			volumeTargetList: &models.ShareTargetList{
+				First: &models.HReference{Href: "https://eu-gb.iaas.cloud.ibm.com/v1/volumes?start=16f293bf-test-4bff-816f-e199c0c65db5\u0026limit=50"},
+				Next:  nil,
+				Limit: 50,
+				ShareTargets: []*models.ShareTarget{
+					{
+						ID:        "16f293bf-test-4bff-816f-e199c0c65db5",
+						MountPath: "abac:/asdsads/asdsad",
+						Name:      "test volume name",
+						Status:    "stable",
+						VPC:       &provider.VPC{ID: "VPC-id1"},
+						ShareID:   "",
+						Zone:      &models.Zone{Name: "test-zone"},
+					},
+				},
+			},
+			expectedReasonCode: "{Trace Code:InternalError, Code:InternalError, Description: Internal Server Error.The file share ID 'volume-id1' could not delete mount target ID '16f293bf-test-4bff-816f-e199c0c65db5'.}",
+			deleteErr:          "Trace Code:InternalError, Code:InternalError, Description: Internal Server Error",
+
+			verify: func(t *testing.T, volumeAccessPointResponse *http.Response, err error) {
+				assert.NotNil(t, volumeAccessPointResponse)
```

Comment:
same here, it will be good if you can avoid using NotNil or Nil

Reply:
We will take this in next PRs when we have more time.